docs/source/en/troubleshooting.mdx

```diff
@@ -173,4 +173,33 @@ tensor([[ 0.0082, -0.2307],
 🤗 Transformers doesn't automatically create an `attention_mask` to mask a padding token if it is provided because:
 
 - Some models don't have a padding token.
-- For some use-cases, users want a model to attend to a padding token.
\ No newline at end of file
+- For some use-cases, users want a model to attend to a padding token.
+
+## ValueError: Unrecognized configuration class XYZ for this kind of AutoModel
+
+Generally, we recommend using the AutoModel class to load pretrained instances of models. This class
+can automatically infer and load the correct architecture from a given checkpoint based on the configuration.
+However, there are a few exceptions to this. In rare cases, some models' architectures don't map to any of the
+AutoModelForXXX classes due to the specifics of their API. When trying to load such an exotic model with the AutoModel
+class, you'll get this `ValueError`.
```

Comment:
There are two situations here:
- the model could be too exotic to be in AutoModel
- but most commonly it's just that the model does not support the task (like there is no GPT2 for question answering for instance)

I think it would be more worthwhile to show an example of the second here than the first.

Reply:
I will add your suggested example, but I think it is still important to mention that exotic model situation can happen, as this can be unexpected. Like in this case, when AutoProcessor works for BLIP-2 but not AutoModel, it can be very confusing for those who don't know how things work under the hood.

Reply:
Updated the example, please take another look